crates/router/src/connector/braintree/transformers.rs

```diff
@@ -254,234 +549,1255 @@ impl From<BraintreePaymentStatus> for enums::AttemptStatus {
+                },
+            },
+        };
+        Ok(Self { query, variables })
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum BraintreeRefundStatus {
+    SettlementPending,
+    Settling,
+    Settled,
+    SubmittedForSettlement,
+    Failed,
+}
+
+impl From<BraintreeRefundStatus> for enums::RefundStatus {
+    fn from(item: BraintreeRefundStatus) -> Self {
+        match item {
+            BraintreeRefundStatus::Settled | BraintreeRefundStatus::Settling => Self::Success,
+            BraintreeRefundStatus::SubmittedForSettlement
+            | BraintreeRefundStatus::SettlementPending => Self::Pending,
+            BraintreeRefundStatus::Failed => Self::Failure,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct BraintreeRefundTransactionBody {
+    pub id: String,
+    pub status: BraintreeRefundStatus,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct BraintreeRefundTransaction {
+    pub refund: BraintreeRefundTransactionBody,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct TransactionError {
-    pub errors: Vec<ErrorObject>,
-    pub credit_card: Option<CreditCardError>,
+pub struct BraintreeRefundResponseData {
+    pub refund_transaction: BraintreeRefundTransaction,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
-pub struct CreditCardError {
-    pub errors: Vec<ErrorObject>,
+#[derive(Debug, Clone, Deserialize, Serialize)]
+
+pub struct RefundResponse {
+    pub data: BraintreeRefundResponseData,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
-pub struct ErrorObject {
-    pub code: String,
-    pub message: String,
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum BraintreeRefundResponse {
+    RefundResponse(Box<RefundResponse>),
+    ErrorResponse(Box<ErrorResponse>),
 }
 
-#[derive(Debug, Default, Deserialize, Serialize)]
+impl TryFrom<types::RefundsResponseRouterData<api::Execute, BraintreeRefundResponse>>
+    for types::RefundsRouterData<api::Execute>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::Execute, BraintreeRefundResponse>,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            response: match item.response {
+                BraintreeRefundResponse::ErrorResponse(error_response) => {
+                    build_error_response(&error_response.errors, item.http_code)
+                }
+                BraintreeRefundResponse::RefundResponse(refund_data) => {
+                    let refund_data = refund_data.data.refund_transaction.refund;
+
+                    Ok(types::RefundsResponseData {
+                        connector_refund_id: refund_data.id.clone(),
+                        refund_status: enums::RefundStatus::from(refund_data.status),
```

Comment:
If 2xx refund error response is a failed refund at the connector end, return a error response for failed status. Use `is_refund_failure` function from connector utils